base/MAPL_AbstractGridFactory.F90

```diff
@@ -183,6 +184,18 @@ subroutine generate_file_bounds(this,grid,local_start,global_start,global_count,
          
       end subroutine generate_file_bounds
 
+      subroutine generate_file_edge_bounds(this,grid,local_start,global_start,global_count,rc)
+         use esmf
+         import AbstractGridFactory
+         class (AbstractGridFactory), intent(inout) :: this
+         type(ESMF_Grid), intent(inout)      :: grid
+         integer, allocatable, intent(inout) :: local_start(:)
+         integer, allocatable, intent(inout) :: global_start(:)
+         integer, allocatable, intent(inout) :: global_count(:)
```

Comment:
I believe these should be `intent(out)`, not `inout`.    With allocatable, the usual concerns that often force us to conservatively use the broader INTENT are not a concern.